src/build123d/topology.py

```diff
@@ -9248,6 +9307,243 @@ def unwrap_topods_compound(
+    """Generic boolean operation for TopoDS_Shapes
+
+    Args:
+        args: Iterable[TopoDS_Shape]:
+        tools: Iterable[TopoDS_Shape]:
+        operation: BRepAlgoAPI_BooleanOperation | BRepAlgoAPI_Splitter:
+
+    Returns: TopoDS_Shape
+
+    """
+
+    arg = TopTools_ListOfShape()
+    for obj in args:
+        arg.Append(obj)
+
+    tool = TopTools_ListOfShape()
+    for obj in tools:
+        tool.Append(obj)
+
+    operation.SetArguments(arg)
+    operation.SetTools(tool)
+
+    operation.SetRunParallel(True)
+    operation.Build()
+
+    result = downcast(operation.Shape())
+    # Remove unnecessary TopoDS_Compound around single shape
+    if isinstance(result, TopoDS_Compound):
+        result = unwrap_topods_compound(result, True)
+
+    return result
+
+
+def _make_topods_face_from_wires(
+    outer_wire: TopoDS_Wire, inner_wires: Optional[Iterable[TopoDS_Wire]] = None
+) -> TopoDS_Face:
+    """_make_topods_face_from_wires
+
+    Makes a planar face from one or more wires
+
+    Args:
+        outer_wire (TopoDS_Wire): closed perimeter wire
+        inner_wires (Iterable[TopoDS_Wire], optional): holes. Defaults to None.
+
+    Raises:
+        ValueError: outer wire not closed
+        ValueError: wires not planar
+        ValueError: inner wire not closed
+        ValueError: internal error
+
+    Returns:
+        TopoDS_Face: planar face potentially with holes
+    """
+    if inner_wires and not BRep_Tool.IsClosed_s(outer_wire):
+        raise ValueError("Cannot build face(s): outer wire is not closed")
+    inner_wires = list(inner_wires) if inner_wires else []
+
+    # check if wires are coplanar
+    verification_compound = _make_topods_compound_from_shapes(
+        [outer_wire] + inner_wires
+    )
+    if not BRepLib_FindSurface(verification_compound, OnlyPlane=True).Found():
+        raise ValueError("Cannot build face(s): wires not planar")
+
+    # fix outer wire
+    sf_s = ShapeFix_Shape(outer_wire)
+    sf_s.Perform()
+    topo_wire = TopoDS.Wire_s(sf_s.Shape())
+
+    face_builder = BRepBuilderAPI_MakeFace(topo_wire, True)
+
+    for inner_wire in inner_wires:
+        if not BRep_Tool.IsClosed_s(inner_wire):
+            raise ValueError("Cannot build face(s): inner wire is not closed")
+        face_builder.Add(inner_wire)
+
+    face_builder.Build()
+
+    if not face_builder.IsDone():
+        raise ValueError(f"Cannot build face(s): {face_builder.Error()}")
+
+    face = face_builder.Face()
+
+    sf_f = ShapeFix_Face(face)
+    sf_f.FixOrientation()
+    sf_f.Perform()
+
+    return TopoDS.Face_s(sf_f.Result())
+
+
+def _sew_topods_faces(faces: Sequence[TopoDS_Face]) -> TopoDS_Shape:
+    """Sew faces into a shell if possible"""
+    shell_builder = BRepBuilderAPI_Sewing()
+    for face in faces:
+        shell_builder.Add(face)
+    shell_builder.Perform()
+    return downcast(shell_builder.SewedShape())
+
+
+def _make_topods_compound_from_shapes(
+    occt_shapes: Sequence[TopoDS_Shape | None],
```

Comment:
```suggestion
    occt_shapes: Iterable[TopoDS_Shape | None],
```
and change all occurrences of
```diff
- _make_topods_compound_from_shapes([...])
+ _make_topods_compound_from_shapes(...)
```

Reply:
Done

Reply:
... and undone.  This change caused strange typing problems where mypy complained about a Sequence when  generator was used even though the type was changed from Sequence to Iterable.  Worth looking at this again in the future.